Google.PowerShell/Logging/GcLogCmdlets.cs

```diff
@@ -145,6 +145,27 @@ protected MonitoredResourceDescriptor GetResourceDescriptor(string descriptorTyp
         protected string[] AllResourceTypes => s_monitoredResourceDescriptors.Value.Select(descriptor => descriptor.Type).ToArray();
 
         /// <summary>
+        /// Generate -ResourceType dynamic parameter.
+        /// </summary>
+        protected RuntimeDefinedParameter GenerateResourceTypeParameter(bool mandatory)
+        {
+            ParameterAttribute paramAttribute = new ParameterAttribute()
+            {
+                Mandatory = mandatory,
+                HelpMessage = "If specified, the cmdlet will filter out log entries based on the resource type."
+            };
+            ValidateSetAttribute validateSetAttribute = new ValidateSetAttribute(AllResourceTypes);
+            validateSetAttribute.IgnoreCase = true;
+            Collection<Attribute> attributes =
```

Comment:
Consider using `var` here, since the code is of the form `TYPE foo = new TYPE(params)`